client/core/core.go

```diff
@@ -1069,9 +1072,12 @@ func (c *Core) dexConnections() []*dexConnection {
 func (c *Core) exchangeMap() map[string]*Exchange {
 	infos := make(map[string]*Exchange, len(c.conns))
 	for _, dc := range c.dexConnections() {
-		dc.cfgMtx.RLock()
-		requiredConfs := uint32(dc.cfg.RegFeeConfirms)
-		dc.cfgMtx.RUnlock()
+		var requiredConfs uint32
+		if dc.cfg != nil {
+			dc.cfgMtx.RLock()
+			requiredConfs = uint32(dc.cfg.RegFeeConfirms)
+			dc.cfgMtx.RUnlock()
+		}
```

Comment:
Here again, can't check `if dc.cfg != nil` without the `cfgMtx` locked. The `cfgMtx` doesn't protect the fields of the stored `*msgjson.ConfigResult`, it protects the `dexConnection.cfg` field itself.